corehq/apps/domain/views.py

```diff
@@ -131,11 +131,16 @@ def select(request, domain_select_template='domain/select.html', do_not_redirect
     else:
         domain = Domain.get_by_name(last_visited_domain)
         if domain and domain.is_active:
-            try:
-                from corehq.apps.dashboard.views import dashboard_default
-                return dashboard_default(request, last_visited_domain)
-            except Http404:
-                pass
+            if (
+                request.couch_user.is_member_of(domain) or domain.is_public
+                or (request.user.is_superuser and not domain.restrict_superusers)
+                or domain.is_snapshot
```

Comment:
was this copied from somewhere or did you just come up with this logic? (seems right, just checking and wondering if it should be abstracted out to a function)